File: 0001-two-sum/0001-two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
 static int a[2];
 *returnSize=2;
 for(int i=0;i<numsSize;i++)
 {
     for(int j=i+1;j<numsSize;j++)
     {
         if(*(nums+i)+*(nums+j)==target)
         {
             a[0]=i;
             a[1]=j;
             break;
         }
     }
 }   
 return a;
}

File: 0001-two-sum/0001-two-sum.java
class Pair{
    int value,index;
    public Pair(int v,int i)
    {
        this.value=v;
        this.index=i;
    }
}


class Solution {
    public int[] twoSum(int[] nums, int target) {
        Pair p[]=new Pair[nums.length];
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            p[i]=new Pair(nums[i],i);
        }
        Arrays.sort(p,(a,b) -> Integer.compare(a.value,b.value));
        int l=0,r=n-1;
        while(l!=r)
        {
            if(target == p[l].value + p[r].value) 
                return new int[] {p[l].index,p[r].index};
            else if(target < p[l].value + p[r].value)
                --r;
            else
                l++;
        }
        return new int[]{};
    }
}